spec/latest/core/core.jsonld

```diff
@@ -505,6 +511,55 @@
       "domain": "hydra:IriTemplateMapping",
       "range": "xsd:string",
       "vs:term_status": "testing"
+    },
+    {
+      "@id": "hydra:returnsHeader",
+      "@type": "rdf:Property",
+      "label": "returns header",
+      "comment": "Name of the header returned by the operation.",
+      "domain": "hydra:Operation",
+      "range": "xsd:string",
+      "vs:term_status": "testing"
+    },
+    {
+      "@id": "hydra:expectsHeader",
+      "@type": "rdf:Property",
+      "label": "expects header",
+      "comment": "Specification of the header expected by the operation.",
+      "domain": "hydra:Operation",
+      "rangeIncludes": ["hydra:Header", "hydra:HeaderTemplate"],
+      "vs:term_status": "testing"
+    },
+    {
+      "@id": "hydra:Template",
+      "@type": "rdfs:Class",
+      "label": "Template",
+      "comment": "Some abstract template.",
+      "vs:term_status": "testing"
+    },
+    {
+      "@id": "hydra:Header",
+      "@type": "rdfs:DataType",
+      "label": "Header",
+      "comment": "Specification of the header, including name and value(s).",
+      "vs:term_status": "testing"
+    },
+    {
+      "@id": "hydra:HeaderTemplate",

```

Comment:
Still on the subject of headers, I think we're reinventing the wheel here.

What do you think about adopting the structure and terminology from actual [HTTP spec](https://tools.ietf.org/html/rfc7230#section-3.2) to describe the headers?

Reply:
This has been discussed already: https://github.com/HydraCG/Specifications/issues/109

> It seems that using unmaintained vocabulary that would in addition tightly couple hydra to a specific protocol is not a good idea. Hydra is meant to be protocol-agnostic, thus I don't see any option of using such vocabularies in the spec. -- https://github.com/HydraCG/Specifications/issues/109#issuecomment-452431098

Regarding this, should we support the description of (HTTP) Headers anyway?

Reply:
Sorry, I did not mean to propose the HTTP vocabulary. Just to follow the actual protocol specification to describe headers

Reply:
Sorry, misinterpreted that, should have followed the link ^^

Reply:
The idea with template in headers was to allow server to provide i.e. complete request template, including headers and/or body so the client just binds that template with variable values and sends back.